core/include/array/array_sorted_read_state.h

```diff
@@ -201,17 +202,14 @@ class ArraySortedReadState {
   /** AIO counter. */
   int aio_cnt_;
 
-  /** The AIO mutex conditions (one for each buffer). */
-  pthread_cond_t aio_cond_[2];
+  /** Condition variables used in AIO. */
+  std::condition_variable aio_cv_[2];
```

Comment:
So here the AIO is used to overlap IO with CPU, effectively hiding one of the two.